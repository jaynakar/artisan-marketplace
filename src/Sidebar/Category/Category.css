.sidebar-title {
    font-size: 14px;
    font-weight: normal;
    margin-bottom: 14px;
    color: black;
  }
  
  .sidebar-items {
    margin-top: 14px;
  }
  
  .sidebar-label-container {
    display: block;
    position: relative;
    padding-left: 24px;
    margin-bottom: 8px;
    cursor: pointer;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
  }
  
  .sidebar-label-container input {
    position: absolute;
    opacity: 0;
    cursor: pointer;
  }
  
  .checkmark {
    position: absolute;
    top: 0;
    left: 0;
    height: 16px;
    width: 16px;
    background-color: #eee;
    border-radius: 50%;
  }
  
  .all {
    background: linear-gradient(blue, crimson);
  }
  
  .sidebar-label-container:hover input ~ .checkmark {
    background-color: #ccc;
  }
  
  .sidebar-label-container input:checked ~ .checkmark {
    background-color: #2196f3;
  }
  
  .checkmark:after {
    content: "";
    position: absolute;
    display: none;
  }
  
  .sidebar-label-container input:checked ~ .checkmark:after {
    display: block;
  }
  
  .sidebar-label-container .checkmark:after {
    top: 6.4px;
    left: 6.4px;
    width: 7px;
    height: 7px;
    border-radius: 50%;
    background: white;
  }
  
  
  .line {
    margin-top: 1.5rem;
    border-color: #f7f7f7;
  }