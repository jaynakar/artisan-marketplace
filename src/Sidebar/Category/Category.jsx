import "./Category.css";
import Input from "../../components/BuyerHomeInput.jsx";
import React from "react";

function Category({ handleChange, selectedValue }) {
  return (
    <section className="category-section" aria-labelledby="category-heading">
      <h2 id="category-heading" className="sidebar-title">Category</h2>
      <fieldset className="category-options">
        <legend className="sr-only">Select an item category</legend>
        <label className="sidebar-label-container">
          <input 
            onChange={handleChange} 
            type="radio" 
            value="" 
            name="category" 
            checked={selectedValue === ""}
          />
          <em className="checkmark" aria-hidden="true"></em>
          All
        </label>
        <Input
          handleChange={handleChange}
          value="Ceramics"
          title="Ceramics"
          name="category"
          checked={selectedValue === "Ceramics"}
        />
        <Input
          handleChange={handleChange}
          value="Jewelry"
          title="Jewelry"
          name="category"
          checked={selectedValue === "Jewelry"}
        />
        <Input
          handleChange={handleChange}
          value="textile"
          title="Textile"
          name="category"
          checked={selectedValue === "textile"}
        />
        <Input
          handleChange={handleChange}
          value="woodwork"
          title="Woodwork"
          name="category"
          checked={selectedValue === "woodwork"}
        />
        <Input
          handleChange={handleChange}
          value="other"
          title="other"
          name="category"
          checked={selectedValue === "other"}
        />
      </fieldset>
    </section>
  );
}

export default Category;