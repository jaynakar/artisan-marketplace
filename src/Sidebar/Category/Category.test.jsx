import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Category from './Category';

describe('Category Component', () => {
  const mockHandleChange = jest.fn();
  const selectedValue = 'Ceramics';

  beforeEach(() => {
    render(
      <Category 
        handleChange={mockHandleChange} 
        selectedValue={selectedValue} 
      />
    );
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders without crashing', () => {
    expect(screen.getByRole('region')).toBeInTheDocument();
  });

  it('has proper accessibility attributes', () => {
    expect(screen.getByRole('region')).toHaveAttribute('aria-labelledby', 'category-heading');
    expect(screen.getByText('Category')).toHaveAttribute('id', 'category-heading');
    expect(screen.getByText('Select an item category')).toHaveClass('sr-only');
  });

  it('renders all category options', () => {
    expect(screen.getByLabelText('All')).toBeInTheDocument();
    expect(screen.getByLabelText('Ceramics')).toBeInTheDocument();
    expect(screen.getByLabelText('Jewelry')).toBeInTheDocument();
    expect(screen.getByLabelText('Textile')).toBeInTheDocument();
    expect(screen.getByLabelText('Woodwork')).toBeInTheDocument();
    expect(screen.getByLabelText('other')).toBeInTheDocument();
  });

  it('shows the correct option as checked based on selectedValue', () => {
    expect(screen.getByLabelText('Ceramics')).toBeChecked();
    expect(screen.getByLabelText('All')).not.toBeChecked();
    expect(screen.getByLabelText('Jewelry')).not.toBeChecked();
  });

  it('calls handleChange when an option is clicked', () => {
    fireEvent.click(screen.getByLabelText('Jewelry'));
    expect(mockHandleChange).toHaveBeenCalledTimes(1);
  });

  it('renders the Input components with correct props', () => {
    const ceramicsInput = screen.getByLabelText('Ceramics');
    expect(ceramicsInput).toHaveAttribute('value', 'Ceramics');
    expect(ceramicsInput).toHaveAttribute('name', 'category');
    expect(ceramicsInput).toBeChecked();
  });

  it('renders the custom radio button correctly', () => {
    const allRadio = screen.getByLabelText('All');
    expect(allRadio).toHaveAttribute('type', 'radio');
    expect(allRadio).toHaveAttribute('value', '');
    expect(allRadio.nextSibling).toHaveClass('checkmark');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <Category 
        handleChange={mockHandleChange} 
        selectedValue={selectedValue} 
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});