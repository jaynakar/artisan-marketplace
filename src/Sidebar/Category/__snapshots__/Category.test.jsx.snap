// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Component matches snapshot 1`] = `
<DocumentFragment>
  <section
    aria-labelledby="category-heading"
    class="category-section"
  >
    <h2
      class="sidebar-title"
      id="category-heading"
    >
      Category
    </h2>
    <fieldset
      class="category-options"
    >
      <legend
        class="sr-only"
      >
        Select an item category
      </legend>
      <label
        class="sidebar-label-container"
      >
        <input
          name="category"
          type="radio"
          value=""
        />
        <em
          aria-hidden="true"
          class="checkmark"
        />
        All
      </label>
      <label
        class="sidebar-label-container"
      >
        <input
          checked=""
          name="category"
          type="radio"
          value="Ceramics"
        />
        <em
          aria-hidden="true"
          class="checkmark"
        />
        Ceramics
      </label>
      <label
        class="sidebar-label-container"
      >
        <input
          name="category"
          type="radio"
          value="Jewelry"
        />
        <em
          aria-hidden="true"
          class="checkmark"
        />
        Jewelry
      </label>
      <label
        class="sidebar-label-container"
      >
        <input
          name="category"
          type="radio"
          value="textile"
        />
        <em
          aria-hidden="true"
          class="checkmark"
        />
        Textile
      </label>
      <label
        class="sidebar-label-container"
      >
        <input
          name="category"
          type="radio"
          value="woodwork"
        />
        <em
          aria-hidden="true"
          class="checkmark"
        />
        Woodwork
      </label>
      <label
        class="sidebar-label-container"
      >
        <input
          name="category"
          type="radio"
          value="other"
        />
        <em
          aria-hidden="true"
          class="checkmark"
        />
        other
      </label>
    </fieldset>
  </section>
</DocumentFragment>
`;
